cmd/srcd/cmd/components.go

```diff
@@ -60,30 +60,56 @@ var componentsInstallCmd = &cobra.Command{
 	Use:   "install [component]",
 	Short: "Install source{d} component",
 	Args:  cobra.MinimumNArgs(1),
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmps, err := components.List(context.Background(), false)
+		if err != nil {
+			return fmt.Errorf("could not list images: %s", err)
+		}
+
 		for _, arg := range args {
-			ok, err := components.IsInstalled(context.Background(), arg)
-			if err != nil {
-				if err == components.ErrNotSrcd {
-					log.Printf("can't install %s, docker image from unknown organization", arg)
-				} else {
-					log.Printf("could not check if %s is installed: %v", arg, err)
+			var c *components.Component
+			for _, cmp := range cmps {
+				// We allow to match by container name or by image name
+				if arg == cmp.Name || arg == cmp.Image || arg == cmp.ImageWithVersion() {
+					c = &cmp
+					break
 				}
-				os.Exit(1)
 			}
 
-			if ok {
-				log.Printf("%q is already installed", arg)
+			if c == nil {
+				names := make([]string, len(cmps))
+				for i, cmp := range cmps {
+					names[i] = cmp.Image
+				}
+
+				return fmt.Errorf("%s is not valid. Component must be one of [%s]", arg, strings.Join(names, ", "))
```

Comment:
This might create a huge error message, maybe it's worth saying it's not an authorized image and how to find the list of valid ones?

Reply:
I don't think it's that big, compared to the huge help that follows below it, and it saves an extra step for the user to find that list:
```
$ ./srcd components install nope
Error: nope is not valid. Component must be one of [srcd/cli-daemon, srcd/gitbase, srcd/gitbase-web, bblfsh/bblfshd, bblfsh/web]
Usage:
  srcd components install [component] [flags]

Flags:
  -h, --help   help for install

Global Flags:
      --config string   config file (default is $HOME/.srcd.yaml)
  -v, --verbose         if true, log all of the things

nope is not valid. Component must be one of [srcd/cli-daemon, srcd/gitbase, srcd/gitbase-web, bblfsh/bblfshd, bblfsh/web]
```

Where would you propose to put the list? The online docs? The `-h` output?

Reply:
Actually, the error is not that bad, indeed.

I'm not sure we should be returning the error since it causes the whole `usage` message to be printed too.

Reply:
Yes, I also dislike the whole `--help` output for a wrong parameter, but it's consistent with other `src-d` commands that use https://github.com/src-d/go-cli.